### DISCUSSION TITLE
Understand the group theory explanation of Dirac point in graphene with the help of `SpaceGroupIrep`.

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [goodluck1982/SpaceGroupIrep#27] [General]
By the help of `SpaceGroupIrep`, I try to understand the blog [here](https://blog.sciencenet.cn/blog-1502061-1201748.html), titled "石墨烯Dirac点的群论解释", as shown below: 

![image](https://user-images.githubusercontent.com/11155854/152670281-4f0cc897-bb1a-44f5-97b7-cb0546ca01bb.png)

I've obtained the SGIrepTab as follows:

```mathematica
<< SpaceGroupIrep`;
showSGIrepTab[191, {0, 0, 0}]
```
 
![image](https://user-images.githubusercontent.com/11155854/152670460-ac65af60-c57f-4c9a-8238-ef26094f313c.png)

But I still can't deduce the specific representation and conclusion presented in the above blog.

Any hints will be highly appreciated.

Regards,
HZ

**Comment by goodluck1982** [comment]
For this problem, using `showLGIrepTab[191, {0, 0, 0}]` is enough.
In fact, the effect of the above command is similar to the following character table. It provide a reference to determine which irep the state/orbital belongs to.

![图片](https://user-images.githubusercontent.com/48718961/152674180-785e8c0e-b383-42c8-ab17-174bec734e86.png)

However, the last line of the above table have to be obtained by the user's own analyses according to their specific problem. Here, it is to apply the operations in the little group of Gamma on the two Bloch states (or bases) to obtain the $\chi_\Gamma$.

If VASP bands are used, one can also use `vasp2trace` and  `getBandRep` to obtain the ireps.

However, the following equation may be the result of a TB model, which cannot be obtained directly from merely group theory.
![图片](https://user-images.githubusercontent.com/48718961/152674135-47483a30-eaa2-4ad4-8b0c-09874f1f8923.png)

**Comment by hongyi-zhao** [comment]
> Here, it is to apply the operations in the little group of Gamma 

There are 12 irreducible representations and 12 conjugate classes belonging to Gamma point. So, how to know which operations in the little group of Gamma are selected and applied?

> on the two Bloch states (or bases) 

Why must there be two Bloch states (or bases) here?

> to obtain the $\chi_\Gamma$.

If I understand correctly, the representation of $\chi_\Gamma$ has the dimension 2, which can be seen from the $E$ column of the  character table, as indicated below: 

![image](https://user-images.githubusercontent.com/11155854/152682824-61623a65-bb89-4a31-b0e3-88fb87a4a94e.png)

> If VASP bands are used, one can also use `vasp2trace` and `getBandRep` to obtain the ireps.

I think you mean the following example shown in [document and examples.nb](https://github.com/goodluck1982/SpaceGroupIrep/blob/main/examples/document%20and%20examples.nb):

![image](https://user-images.githubusercontent.com/11155854/152683209-0fc22b5b-8c05-4bca-9610-5ea52a159d8d.png)

But I'm still a little confused. Take the following screenshot as an example:

![image](https://user-images.githubusercontent.com/11155854/152683393-84c1030c-d720-49e5-9a1b-1f024fef99b6.png)

Here, the third column is the energy level degeneracy, and the fourth column inside curly braces is the determined LGIR, which is composed of the extended Mulliken symbol and the Gamma notation with the dimension of the representation denoted in parentheses.

But the following notation is not consistent with the Mulliken symbol E_1 and E_2 in writing styles:

![image](https://user-images.githubusercontent.com/11155854/152683784-4cb4c127-f994-4285-8390-19c3cae6099f.png)

OTOH, both E_1 and E_2 are 2 dimensional representations, the direct sum of them should give a 2+2 = 4 dimensional representation. Now, what does this mean physically?

**Comment by goodluck1982** [comment]
> > Here, it is to apply the operations in the little group of Gamma
> 
> There are 12 irreducible representations and 12 conjugate classes belonging to Gamma point. So, how to know which operations in the little group of Gamma are selected and applied?

In principle, ALL operations should be used. But in practice, maybe after you have used several operations, the result can be determined, and then you can stop.

> 
> > on the two Bloch states (or bases)
> 
> Why must there be two Bloch states (or bases) here?

Because it is said that there are two pz orbitals.
![图片](https://user-images.githubusercontent.com/48718961/152685311-dbaa645b-afcf-42fa-8882-45a9e1047694.png)
And, OTOH, ![图片](https://user-images.githubusercontent.com/48718961/152685353-bfb6a848-7289-4d48-b3a4-1529d9deb180.png) this can tell us there are only two bases to give a 2-dimensional representation.

> > to obtain the $\chi_\Gamma$.
> 
> If I understand correctly, the character of $\chi_\Gamma$ is 2, which must be equal to the characters' summation of the direct sum factors representations, as indicated below:
> 
> > If VASP bands are used, one can also use `vasp2trace` and `getBandRep` to obtain the ireps.
> 
> I think you mean the following example shown in [document and examples.nb](https://github.com/goodluck1982/SpaceGroupIrep/blob/main/examples/document%20and%20examples.nb):
>  
> But I'm still a little confused. Take the following screenshot as an example:
> 
> Here, the third column is the energy level degeneracy, and the fourth column inside curly braces is the determined LGIR, which is composed of the extended Mulliken symbol and the Gamma notation with the dimension of the representation denoted in parentheses.
> 
> But the following notation is not consistent with the Mulliken symbol E_1 and E_2 in writing styles:
> 
> OTOH, both E_1 and E_2 are 2 dimensional representations, the direct sum of them should give a 2+2 = 4 dimensional representation. Now, what does this mean physically?

The original  Mulliken symbol is only defined for point group. Therefore, the BC book uses the  **extended** Mulliken symbol, i.e., the authors extended the  Mulliken symbol. E, E_1, E_2 are all 2D rep. But if this 2D rep is constructed by two 1D reps which are related by time-reversal symmetry, to differentiate the two 1D reps, the superscript 1 or 2 to the left of E is used. Refer to Tab. 5.8 of the BC book:
![图片](https://user-images.githubusercontent.com/48718961/152685835-5cd501a3-5a82-400a-8db5-15591d9d1786.png)

**Comment by hongyi-zhao** [comment]
> > > Here, it is to apply the operations in the little group of Gamma
> > 
> > 
> > There are 12 irreducible representations and 12 conjugate classes belonging to Gamma point. So, how to know which operations in the little group of Gamma are selected and applied?
> 
> In principle, ALL operations should be used. But in practice, maybe after you have used several operations, the result can be determined, and then you can stop.
> 
> > > on the two Bloch states (or bases)
> > 
> > 
> > Why must there be two Bloch states (or bases) here?
> 
> Because it is said that there are two pz orbitals. ![图片](https://user-images.githubusercontent.com/48718961/152685311-dbaa645b-afcf-42fa-8882-45a9e1047694.png) And, OTOH, ![图片](https://user-images.githubusercontent.com/48718961/152685353-bfb6a848-7289-4d48-b3a4-1529d9deb180.png) this can tell us there are only two bases to give a 2-dimensional representation.

The book [ATOMIC and MOLECULAR SYMMETRY GROUPS and CHEMISTRY](https://sg1lib.org/book/15345782/01507f) by S.C. RAKSHIT has the following description in its "APPENDIX II Character Tables of Molecular Symmetry Groups":

![image](https://user-images.githubusercontent.com/11155854/152687079-47772cfa-0070-45d0-ad15-cce97ad0b149.png)

So, if we try to find the pz orbital formed by combinations of irreducible representations, only the ones whose basis vectors and components pertinent to z should be considered. This will greatly reduce the search effort. 

Anyway, this is an algorithmic thing, and shouldn't be a problem solved by hand.

**Comment by goodluck1982** [comment]
Actually, the book 《群论及其在固体物理中的应用》（徐婉棠，喀兴林）also has such character tables with bases. But this works only for point group. If a problem involves energy bands, the group used should be little group, not merely point group, and the bases are actually Bloch waves, not merely the atomic-like orbitals, although in some simple cases using point group can also give correct result.

**Comment by hongyi-zhao** [comment]
If I understand correctly: The only thing in common is that the base is always a linear combination of a specific subset of irreducible representations, which for Bloch waves is consistent with the state superposition principle.

**Comment by goodluck1982** [comment]
Strictly speaking, the bases of little group at k should be the "Bloch sum", i.e. the following psi_jk(r)
![图片](https://user-images.githubusercontent.com/48718961/152727648-e7b23678-3ca3-4063-82d6-0b2b0353c6ee.png)
It has to be the basis of the translation group simultaneously.
As for the local orbitals phi_j, they can be selected from the bases of a point group.

**Comment by hongyi-zhao** [comment]
Thank you for your further detailed description. I give some additional comments below.

This is the so-called linear combination of atomic orbitals (LCAO) method as described [here](https://chempedia.info/page/111066217027067008056159008145008127157221118117/):

![image](https://user-images.githubusercontent.com/11155854/152742353-8ab30dc8-0a31-49ce-a4ca-7e7081cd4d6f.png)

As noted [here](https://en.wikipedia.org/wiki/Basis_set_(chemistry)): The basis set can either be composed of [atomic orbitals](https://en.wikipedia.org/wiki/Atomic_orbital) (yielding the [linear combination of atomic orbitals](https://en.wikipedia.org/wiki/Linear_combination_of_atomic_orbitals) approach), which is the usual choice within the quantum chemistry community; [plane waves](https://en.wikipedia.org/wiki/Plane_wave) which are typically used within the solid state community, such as the ones implemented in VASP, CASTEP, ABINIT, and Quantum ESPRESSO. 

 
Regardless of the basis set, the ultimate goal is to achieve an accurate and efficient approximate of the following famous quantum mechanics identity, as stated [here](https://en.wikipedia.org/wiki/Basis_set_(chemistry)):

$$\left|\psi_{i}\right\rangle \approx \sum_{\mu} c_{\mu i}|\mu\rangle$$

In principle, the results cannot depend on the choice of the basis.

**Comment by goodluck1982** [comment]
> Regardless of the basis set, the ultimate goal is to achieve an accurate and efficient approximate of the following famous quantum mechanics identity, as stated [here](https://en.wikipedia.org/wiki/Basis_set_(chemistry)):
> 
> ![image](https://user-images.githubusercontent.com/11155854/152744509-aece1e4d-f88f-4338-b797-a24ff4e717dd.png)
> 
> In principle, whatever basis set is used, should it give the same or similar result for the same system, as long as it is accurate enough.

Yes, the ultimate goal is to solve the eigenstate |psi_i> you mentioned above, regardless of the basis set.
However, when you do symmetry analyses, the bases are generally important, because different bases sets will result in different representation matrices. Generally, one cannot use the representation matrices obtained otherwhere directly to their own problem. This can be done only when the same bases set is used.

**Comment by hongyi-zhao** [comment]
But the process to solve the eigenstate `|psi_i>` is in nature a series of similarity transformations, which don't change the trace of the matrix used by the representation. So, the `trace.txt` file is needed to do the symmetry analyses under the different target/interested bases.

**Comment by goodluck1982** [comment]
I mean, in some cases, one have to utilize the representation matrices, not only the traces, and in these cases bases are important. Any matrix-form Hamiltonian is defined under a certain bases set, and the symmetry operation matrix (i.e. the representation matrix) operating on the Hamiltonian is also dependent on the bases set.

**Comment by hongyi-zhao** [comment]
This is where I get confused. Do you mean the `trace.txt` file also includes the representation matrices/matrix-form Hamiltonian? Take [SG186-trace.txt](https://github.com/goodluck1982/SpaceGroupIrep/blob/main/examples/SG186-trace.txt) as an example: its data structure is rather complicated, which gives me the impression that there are more information, not just trace.

**Comment by goodluck1982** [comment]
`trace.txt` contains the information of traces, not the bases or representation matrices. But one have to know that the corresponding bases are Bloch functions, not atomic-like local orbitals.

**Comment by hongyi-zhao** [comment]
> `trace.txt` contains the information of traces, not the bases or representation matrices.

Thank you for pointing this out. There is [a (self-explanatory) `trace.txt` file](https://www.cryst.ehu.es/html/cryst/topological/File_Description.txt) which includes the corresponding specification given on [Check Topological Mat.](http://www.cryst.ehu.es/cryst/checktopologicalmat).

> But one have to know that the corresponding bases are Bloch functions, not atomic-like local orbitals.

OMG. I see your point: `trace.txt` is correlated to the underlying bases or representation matrices. So, when the users perform subsequent analysis and research based on the results given by `SpaceGroupIrep`, they must always remember what type of base set `trace.txt` comes from and do the appropriate explanation accordingly.

**Comment by goodluck1982** [comment]
> they must always remember what type of base set `trace.txt` comes from and do the appropriate explanation accordingly.

You are right!
When dealing with energy bands, remember always that the operational objectives of symmetry operations are Bloch waves.

**Comment by hongyi-zhao** [comment]
> The original Mulliken symbol is only defined for point group. Therefore, the BC book uses the **extended** Mulliken symbol, i.e., the authors extended the Mulliken symbol. E, E_1, E_2 are all 2D rep. But if this 2D rep is constructed by two 1D reps which are related by time-reversal symmetry, to differentiate the two 1D reps, the superscript 1 or 2 to the left of E is used. Refer to Tab. 5.8 of the BC book:

According to the following description on page 389 of the BC book:

![image](https://user-images.githubusercontent.com/11155854/152917064-afe2350b-f94c-4d55-9d73-7b4fa86cdc9a.png)

The time-reversal symmetry (TRS) only appears in complex representations. Based on my intuitive understanding, there is a close relationship between TRS and spin.

**Comment by goodluck1982** [comment]
The existence of TRS does not depend on the existence of spin. For a spinless wavefunction \psi(r), its time reversal is just its complex conjugate, i.e. \psi(r)*

**Reply by hongyi-zhao** [reply]
But, there is really a deep connection between spin and time-reversal symmetry (TRS)/time-inverse symmetry (TIS), as described [here](https://staff.aist.go.jp/v.zayets/spin3_51_WhatIsSpin.html):

> The spin describes the properties related to the breaking of the time-inverse symmetry for an elementary particle. An electron has a magnetic moment and angular moment due to the spin (broken time-inverse symmetry).

**Comment by hongyi-zhao** [comment]
Thank you for giving a specific example to illustrate the independence between them. Your example also explains why TRS must appear in the form of paired conjugates: They always come with each other. So it's necessary to make them as a whole in the character table.

**Comment by hongyi-zhao** [comment]
Here's another related research about TRS, which is also the origin of my hunch that there is a close relationship between TRS and spin, as described [here](http://dx.doi.org/10.1103/PhysRevLett.107.066602): 

![image](https://user-images.githubusercontent.com/11155854/153115636-5695dfea-106e-4daa-a4b3-066fe058e4fc.png)

However, as the PRL article above elaborated, the authors declaimed that they have found a QSH-like phase in a system where the TR symmetry is broken. 

The above result is consistent with the description [here](https://en.wikipedia.org/wiki/Quantum_spin_Hall_effect):

![image](https://user-images.githubusercontent.com/11155854/153116587-c815bc42-97b5-40f3-9bd3-74c5856084d9.png)

**Reply by hongyi-zhao** [reply]
TRS plays an important and special role in [Topological Materials](https://cfm.ehu.es/view/files/Thesis_Iñigo_Robredo_final.pdf) and [Valleytronics](https://en.wikipedia.org/wiki/Valleytronics). Below is a description of `Time-Reversal Symmetry and Topology of the Electronic System` from the famous book [`Electronic Structure: Basic Theory and Practical Methods, 2nd Edition` by Richard M. Martin](https://hk1lib.org/book/6037268/3f7c7b):

![image](https://user-images.githubusercontent.com/11155854/157372452-d8ac77ce-9452-4e9f-af1a-bcb819218581.png)
![image](https://user-images.githubusercontent.com/11155854/157372554-2369aed8-4b97-437a-9b46-b75e24e9c317.png)

**Reply by goodluck1982** [reply]
Yes
